packages/wordcount/src/index.js

```diff
@@ -17,18 +17,30 @@ import stripShortcodes from './stripShortcodes';
 import stripSpaces from './stripSpaces';
 import transposeHTMLEntitiesToCountableChars from './transposeHTMLEntitiesToCountableChars';
 
+/**
+ * A number, or a string containing a number.
+ *
+ * @typedef {'words'|'characters_excluding_spaces'|'characters_including_spaces'} WordCountStrategy
+ */
+
+/**
+ * Lower-level settings for word counting that can be overridden.
+ *
+ * @typedef {import("./defaultSettings").WordCountSettings}  WordCountSettings
+ */
+
 /**
  * Private function to manage the settings.
  *
- * @param {string} type         The type of count to be done.
- * @param {Object} userSettings Custom settings for the count.
+ * @param {WordCountStrategy} type The type of count to be done.
+ * @param {WordCountSettings} userSettings Custom settings for the count.
  *
- * @return {void|Object|*} The combined settings object to be used.
+ * @return {WordCountSettings} The combined settings object to be used.
  */
 function loadSettings( type, userSettings ) {
 	const settings = extend( defaultSettings, userSettings );
 
-	settings.shortcodes = settings.l10n.shortcodes || {};
+	settings.shortcodes = settings.l10n?.shortcodes ?? [];
```

Comment:
I'm surprised there is nothing flagged here, since as far as I can tell, `settings.l10n` should be considered required, no? Based on type `Partial<{type: WordCountStrategy, shortcodes: string[]}>`. Unless "Partial" means "possibly undefined", which could be possible but would be new information to me.

Reply:
According to the documentation (and it seems to work like this in practice), Partial is exactly that – [allows for any of the properties to be missing](https://www.typescriptlang.org/v2/docs/handbook/utility-types.html#partialt). I am curious, what did you expect that `Partial` gave us?

Reply:
Oh, I understand that, but based on your usage of optional chaining on `l10n`, it would seem to imply that `settings.l10n` could be `undefined`. I cannot see how that could be the case, based on the type. The `Partial` is relevant for properties within `l10n`, e.g. `settings.l10n.shortcodes` could be omitted.

Reply:
@aduth `WPWordCountSettings` is also a `Partial`:

`@typedef {Partial<WPWordCountSettingsFields>} WPWordCountSettings `

Reply:
> @aduth `WPWordCountSettings` is also a `Partial`:
> 
> `@typedef {Partial<WPWordCountSettingsFields>} WPWordCountSettings `

Ah, I failed to follow that. It makes sense now 👍 I guess this is another example where it's best to let TypeScript manage this and avoid the sort of human error I'm apparently prone to 😄